File: api/src/test/java/net/blueshell/api/factory/dto/TelemetryDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.common.enums.PlatformType;
import net.blueshell.api.dto.TelemetryDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for TelemetryDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class TelemetryDTOFactory extends BaseDtoFactory<TelemetryDTO> {

    @Override
    public Class<TelemetryDTO> targetType() {
        return TelemetryDTO.class;
    }

    @Override
    public TelemetryDTO createBasic() {
        TelemetryDTO dto = new TelemetryDTO();
        dto.setPlatform(PlatformType.FACEBOOK);
        dto.setUrl("https://example.com/" + nextId());
        dto.setCreatedAt(now());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/TelemetryDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.domain.telemetry.web.dto.TelemetryDTO
import net.blueshell.api.shared.enums.PlatformType
import org.springframework.stereotype.Component

/**
 * Factory for TelemetryDTO test instances.
 */
@Component
class TelemetryDTOFactory : BaseDtoFactory<TelemetryDTO>() {

    override fun targetType(): Class<TelemetryDTO> = TelemetryDTO::class.java

    override fun createBasic(): TelemetryDTO {
        val dto = TelemetryDTO()
        dto.platform = PlatformType.FACEBOOK
        dto.url = "https://example.com/${nextId()}"
        dto.createdAt = now()
        return dto
    }
}
